Tools/CISupport/ews-build/steps.py

```diff
@@ -407,9 +407,15 @@ def __init__(self, **kwargs):
     def start(self):
         self.log_observer = logobserver.BufferLogObserver()
         self.addLogObserver('stdio', self.log_observer)
-        revision = self.getProperty('ews_revision', self.getProperty('got_revision'))
-        if not revision:
-            revision = 'HEAD'
+
+        revision = 'HEAD'
+        for prop in ['ews_revision', 'github.base.sha', 'got_revision']:
```

Comment:
Would be good to add a comment here indicating that this list is in specific priority order.